front/components/assistant_builder/AssistantBuilderTablesModal.tsx

```diff
@@ -160,91 +208,44 @@ export default function AssistantBuilderTablesModal({
+            dataSourceView={selectedDataSourceView}
             onPick={(table: CoreAPITable) => {
               const config = {
                 workspaceId: owner.sId,
                 dataSourceId: table.data_source_id,
                 tableId: table.table_id,
                 tableName: table.name,
               };
-              onSave([config], selectedDataSource);
+              onSave([config], selectedDataSourceView);
               onClose();
             }}
             onBack={() => {
-              setSelectedDataSource(null);
+              setSelectedDataSourceOrView(null);
             }}
             tablesQueryConfiguration={tablesQueryConfiguration}
           />
         ) : (
-          <PickTablesManaged
-            owner={owner}
-            dataSource={selectedDataSource}
-            onSelectionChange={(
-              node: ContentNode,
-              parents: string[],
-              selected: boolean
-            ) => {
-              setInternalIds((internalIds) => {
-                const newIds = internalIds.filter(
-                  (id) => id !== node.internalId
-                );
-                const newNodes = selectedManagedTables.filter(
-                  (n) => n.internalId !== node.internalId
-                );
-                const newParentsById = Object.entries(
-                  parentsById as Record<string, Set<string>>
-                ).reduce(
-                  (acc, [key, value]) =>
-                    key === node.internalId
-                      ? acc
-                      : {
-                          ...acc,
-                          [key]: value,
-                        },
-                  {} as Record<string, Set<string>>
-                );
-
-                if (selected) {
-                  newIds.push(node.internalId);
-                  newNodes.push(node);
-                  newParentsById[node.internalId] = new Set(
-                    // This is to get the same structure/order in the fallback as the endpoint return, from leaf to root, including leaf.
-                    [...parents, node.internalId].reverse()
-                  );
-                }
-                // Optimistic update
-                const key = serializeUseDataSourceKey({
-                  workspaceId: owner.sId,
-                  dataSourceName: selectedDataSource.name,
-                  internalIds: newIds,
-                });
-                setFallback((prev) => ({
-                  ...prev,
-                  [key]: {
-                    contentNodes: newNodes,
-                    parentsById: newParentsById,
-                  },
-                }));
-                return newIds;
-              });
-            }}
-            selectedNodes={selectedManagedTables}
-            onBack={() => {
-              setSelectedDataSource(null);
-            }}
-            parentsById={parentsById || {}}
-          />
+          selectedDataSourceView && (
```

Comment:
Why not allowing `PickTablesManaged` to take a nullable `selectedDataSourceView` and return null from there?